src/main/java/uk/gov/companieshouse/overseasentitiesapi/controller/OverseasEntitiesDataController.java

```diff
@@ -109,4 +118,48 @@ private ResponseEntity<OverseasEntityDataApi> getOverseasEntityDataResponse(
         
         return ResponseEntity.ok(overseasEntityDataApi);
     }
+
+    @GetMapping("/beneficial-owners")
+    public ResponseEntity<PrivateBoDataListApi> getOverseasEntityBeneficialOwners(
+            @PathVariable(TRANSACTION_ID_KEY) String transactionId,
+            @PathVariable(OVERSEAS_ENTITY_ID_KEY) String overseasEntityId,
+            @RequestHeader(value = ERIC_REQUEST_ID_KEY) String requestId) throws ServiceException, NoSuchAlgorithmException {
+
+        final var logMap = new HashMap<String, Object>();
+        logMap.put(OVERSEAS_ENTITY_ID_KEY, overseasEntityId);
+        logMap.put(TRANSACTION_ID_KEY, transactionId);
+        ApiLogger.infoContext(requestId, "Calling service to retrieve private beneficial owner information", logMap);
+
+        isRoeUpdateFlagEnabled();
+        final Optional<OverseasEntitySubmissionDto> overseasEntitySubmissionDto = overseasEntitiesService.getOverseasEntitySubmission(overseasEntityId);
+        if(overseasEntitySubmissionDto.isPresent() && overseasEntitySubmissionDto.get().isForUpdate()) {
+            String entityNumber = overseasEntitySubmissionDto.get().getEntityNumber();
+            try {
+                PrivateBoDataListApi privateBeneficialOwnersData = privateDataRetrievalService.getBeneficialOwnersData(entityNumber);
+
+                if (privateBeneficialOwnersData == null || privateBeneficialOwnersData.getBoPrivateData().isEmpty()) {
+                    final var message = String.format("Beneficial owner private data not found for overseas entity %s",
+                            overseasEntityId);
+                    ApiLogger.errorContext(requestId, message, null, logMap);
+                    return ResponseEntity.notFound().build();
+                }
+
+                var hashHelper = new HashHelper(salt);
+                for (PrivateBoDataApi privateBoData : privateBeneficialOwnersData) {
+                    var hashedId = hashHelper.generateHashedId(privateBoData.getPscId());
+                    privateBoData.setHashedId(hashedId);
```

Comment:
should we set PscId to be null or ""? May need annotation on JSON model for null.
THat would stop the Ids leaking out.

Reply:
private sdk repo JSON model has been updated

Reply:
great, just need to set pscId to null here with privateBoData.setPscId(null);